packages/swing-store/src/swingStore.js

```diff
@@ -82,6 +87,154 @@ export function makeSnapStoreIO() {
  * }} SwingStore
  */
 
+/**
+ * @typedef {[
+ *   key: string,
+ *   value: string|null,
+ * ]} KVPair
+ *
+ * @typedef {object} SwingStoreExporter
+ *
+ * Allows export of data from a swingStore as a fixed view onto the content as
+ * of the most recent commit point at the time the exporter was created.  The
+ * exporter may be used while another SwingStore instance is active for the same
+ * DB, possibly in another thread or process.  It guarantees that regardless of
+ * the concurrent activity of other swingStore instances, the data representing
+ * the commit point will stay consistent and available.
+ *
+ * @property {() => Iterable<KVPair>} getKVData
+ *
+ * Get a full dump of KV data from the swingStore. This represents both the
+ * KVStore (excluding host and local prefixes), as well as any data needed to
+ * validate all artifacts, both current and historical. As such it represents
+ * the root of trust for the application.
+ *
+ * Likely content of validation data (with supporting entries for indexing):
+ * - export.snapshot.${vatID}.${endPos} = ${{ vatID, endPos, hash, uncompressedSize, compressedSize });
+ * - export.lastSnapshot.${vatID} = ${endPos}
+ * - export.transcript.${vatID}.${startPos} = ${{ vatID, startPos, endPos, hash, uncompressedSize, compressedSize }}
+ *
+ * @property {(includeHistorical: boolean) => Iterable<string>} getArtifactNames
```

Comment:
Let's keep this flexible and make this async.

```suggestion
 * @property {(includeHistorical: boolean) => AsyncIterable<string>} getArtifactNames
```

Reply:
All the ways this data are produced are synchronous, so what does this buy us aside from forcing inconvenient `await`s (and all the attendant folderol that entrains) onto clients?

Reply:
This is an interface that can be implemented by other components than the SQLite backed swingstore. Who is to say we may not have a way to stream the names of available artifacts on some remote archive node in the future. The only clients are the swingstore importer and the host app, which are already async, and to appease the eslint rules asking for no await in for loops, may do `for await` on this iterator anyway.

Reply:
This same future-proofing argument could in principle apply to all interfaces everywhere.  I remain unconvinced.

Reply:
I agree with Chip, a consumer which needs to be async can accept a `getExportData`/`getArtifactNames`/`getArtifact` API which is either sync or async. And making these export-side APIs sync allows callers to be sync too, if they want.

The *import* side should accept both sync and async providers, which then constrains the top-level import API to be async. But I don't see a reason why we should impose gratuitious asynchronousness on an *export* API that doesn't need it.

If a future implementation of this API cannot provide a synchronous one, that's ok, it will limit the clients of that API to be async themselves, but that's the future's business. We can introduce a new type later (`AsyncSwingStoreExporter`?).

Reply:
Maybe I'm not expressing myself correctly enough.

The host consumer and the swingstore importer are both already async, they needs to call `getArtifact` which can only be an AsyncIterable, so we're not imposing any additional async complexity on the consuming implementation.

`SwingStoreExporter` is an interface which is implemented by both the SwingStore and by the host to be used by the SwingStore importer. While the SwingStore implementation is sync today, and the host implementation will likely be sync as well, I would like to make sure we don't need to change the implementation of the SwingStore importer (aka change a simple `for (const name of exporter.getArtifactNames())` to `for await (const name of exporter.getArtifactNames())`) if we need to change the host implementation to be async.